src/main/java/io/quarkus/search/app/entity/I18nData.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.search.app.entity;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
+import jakarta.persistence.Embeddable;
+
+@Embeddable
+public class I18nData<T> {
+    @JdbcTypeCode(SqlTypes.JSON)
+    Map<String, T> data = new LinkedHashMap<>();
```

Comment:
Storing JSON in a DB makes me scream internally, but if you think it's better in this case... ok.

Reply:
And then searching over these JSON fields inside a DB 😈 😄 
But no, I hoped that it just be simpler at the time to use JSON rather than overriding JPA attributes; that I did anyway in the end 😄